tests/integration/test_keeper_internal_secure/test.py

```diff
@@ -61,23 +72,113 @@ def get_fake_zk(nodename, timeout=30.0):
-        node1_zk = get_fake_zk("node1")
-        node2_zk = get_fake_zk("node2")
-        node3_zk = get_fake_zk("node3")
+        for node in nodes:
+            node_zks.append(get_fake_zk(node.name))
 
-        node1_zk.create("/test_node", b"somedata1")
-        node2_zk.sync("/test_node")
-        node3_zk.sync("/test_node")
+        node_zks[0].create("/test_node", b"somedata1")
+        node_zks[1].sync("/test_node")
+        node_zks[2].sync("/test_node")
 
-        assert node1_zk.exists("/test_node") is not None
-        assert node2_zk.exists("/test_node") is not None
-        assert node3_zk.exists("/test_node") is not None
+        for node_zk in node_zks:
+            assert node_zk.exists("/test_node") is not None
     finally:
         try:
-            for zk_conn in [node1_zk, node2_zk, node3_zk]:
+            for zk_conn in node_zks:
+                if zk_conn is None:
+                    continue
                 zk_conn.stop()
                 zk_conn.close()
         except:
             pass
+
+
+def setupSsl(node, filename, password):
+    if password is None:
+        node.copy_file_to_container(
+            os.path.join(CURRENT_TEST_DIR, "configs/ssl_conf.yml"),
+            "/etc/clickhouse-server/config.d/ssl_conf.yml",
+        )
+
+        node.replace_in_config(
+            "/etc/clickhouse-server/config.d/ssl_conf.yml",
+            "WithoutPassPhrase",
+            filename,
+        )
+        return
+
+    node.copy_file_to_container(
+        os.path.join(CURRENT_TEST_DIR, "configs/ssl_conf_password.yml"),
+        "/etc/clickhouse-server/config.d/ssl_conf.yml",
+    )
+
+    node.replace_in_config(
+        "/etc/clickhouse-server/config.d/ssl_conf.yml",
+        "WithoutPassPhrase",
+        filename,
+    )
+
+    node.replace_in_config(
+        "/etc/clickhouse-server/config.d/ssl_conf.yml",
+        "PASSWORD",
+        password,
+    )
+
+
+def stop_all_clickhouse():
+    for node in nodes:
+        node.stop_clickhouse()
+
+    for node in nodes:
+        node.exec_in_container(["rm", "-rf", "/var/lib/clickhouse/coordination"])
+
+
+def start_clickhouse(node):
+    node.start_clickhouse()
+
+
+def start_all_clickhouse():
+    p = Pool(3)
+    waiters = []
+
+    for node in nodes:
+        waiters.append(p.apply_async(start_clickhouse, args=(node,)))
+
+    for waiter in waiters:
+        waiter.wait()
+
+    for node in nodes:
+        ku.wait_until_connected(cluster, node)
+
+
+def check_valid_configuration(filename, password):
+    stop_all_clickhouse()
+    for node in nodes:
+        setupSsl(node, filename, password)
+    start_all_clickhouse()
+    run_test()
+
+
+def check_invalid_configuration(filename, password):
+    stop_all_clickhouse()
+    for node in nodes:
+        setupSsl(node, filename, password)
+
+    nodes[0].start_clickhouse(expected_to_fail=True)
+    nodes[0].wait_for_log_line(
+        "OpenSSLException: EVPKey::loadKey.*error:0480006C:PEM routines::no start line",
```

Comment:
In 24.3 we are still using BoringSSL, so the error message is a bit different:
`OpenSSLException: EVPKey::loadKey(string): error:0900006e:PEM routines:OPENSSL_internal:NO_START_LINE`